### PR TITLE
Add None transform support to Datasets

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
Continuous task for #717 .
Currently, the Datasets can't support None transform, this PR added support for that.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

File: monai/data/dataset.py [hunk 1]
```diff
@@ -237,7 +239,12 @@ class CacheDataset(Dataset):
     """
 
     def __init__(
-        self, data, transform: Callable, cache_num: int = sys.maxsize, cache_rate: float = 1.0, num_workers: int = 0
+        self,
```
Comment (wyli) [inline monai/data/dataset.py:242]: I think the cache dataset is to accelerate the transforms, supporting None transform is quite confusing. why do we need this?

Comment (Nic-Ma) [reply monai/data/dataset.py:242]: Because some other datasets are based on this CacheDataset, if no transforms, it should also work fine, right?
Just like `cache_num = 0`.
What do you think?

Comment (Nic-Ma) [reply monai/data/dataset.py:242]: And all the other datasets in MONAI can support None transform, I think we should keep the same logic for them?

Comment (wyli) [reply monai/data/dataset.py:242]: IMO the high-level dataset as an extension of these basic datasets should handle these cases. by default the basic cache datasets should require a user-specified transform. and it seems the current CacheDataset works fine if the user input to transform is None

Comment (wyli) [reply monai/data/dataset.py:242]: > And all the other datasets in MONAI can support None transform, I think we should keep the same logic for them?
support None and default to None are different here...

Comment (Nic-Ma) [reply monai/data/dataset.py:242]: OK, I see.
I didn't realize that the `Compose` generates an empty list for None:
```py
if transforms is None:
    transforms = []
```
I will revert the modification to `CacheDataset` and only add the unit tests for None transform.
And I will also remove the `None` default value for `PersistentDataset`.

Thanks.

Comment (wyli) [reply monai/data/dataset.py:242]: thank you, I think the unit tests for None transforms looks nice

Comment (Nic-Ma) [reply monai/data/dataset.py:240]: updated in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review.
I updated the PR according to your comments.
Could you please help review it again?
Thanks.
